Open_Source/Section-5/5.38_Explain_everything_related_to_Flux.md

```diff
@@ -0,0 +1,113 @@
+
+`Action creators — dispatcher helper methods` — are used to support a semantic API that describes all changes that are possible in the application. It can be useful to think of them as a fourth part of the Flux update cycle
+
+## How does the API relate to this?
+
+When you are working with data that is coming from (or going to) the outside, I’ve found that using Actions to introduce the data into the Flux Flow, and subsequently Stores, is the most painless way to go about it.
+
+## Action Creators & Actions
+
+**Action Creators** are collections of methods that are called within views (or anywhere else for that matter) to send actions to the Dispatcher. Actions are the actual payloads that are delivered via the dispatcher.
+
+The way Facebook uses them, action type constants are used to define what action should take place, and are sent along with action data. Inside of `registered callbacks`, these actions can now be handled according to their action type, and methods can be `called` with action data as the arguments.
+
+`// ES5`</br>
+`var keyMirror = require('react/lib/keyMirror');`</br>
+`module.exports = keyMirror({`</br>
+  `LOAD_SHOES: null`</br></br>
+`});`</br>
+`// ES6`</br>
+`import keyMirror from 'react';`</br>
+`export keyMirror({`</br>
+        `LOAD_SHOES: null`</br>
+   ` });`</br>
+
+Above we use React’s _**keyMirror library**_ to mirror our keys so that our value matches our key definition.
+Just by looking at this file, we can tell that our app loads shoes. The use of constants helps keep things organized, and helps give a high level view of what the app actually does.
+Now lets take a look at the corresponding Action Creator definition:
+
+`// ES 5`</br>
+`var AppDispatcher = require('../dispatcher/AppDispatcher');`</br>
+`var ShoeStoreConstants = require('../constants/ShoeStoreConstants');`</br>
+`var ShoeStoreActions = {`</br>
+    `loadShoes: function(data) {`</br>
+   ` AppDispatcher.handleAction({`</br>
+     ` actionType: ShoeStoreConstants.LOAD_SHOES,`</br>
+     ` data: data`</br>
+    `})`</br>
+  `}`</br>
+`};`</br>
+`module.exports = ShoeStoreActions;`</br>
+`// ES6`</br>
+`import AppDispatcher from '../dispatcher/AppDispatcher';`</br>
+`import ShoeStoreConstants from '../constants/ShoeStoreConstants';`</br>
+`class ShoeStoreActions {`</br>
+    `public function loadShoes(data) {`</br>
+        `AppDispatcher.handleAction({`</br>
+          `actionType: ShoeStoreConstants.LOAD_SHOES,`</br>
+          `data: data`</br>
+        `});`</br>
+        `return;`</br>
+    `}`</br>
+`}`</br>
+In our example above, we created a method on our _ShoeStoreActions_ object that calls our _dispatcher_ with the data we provided. We can now import this actions file into our view or API, and call _ShoeStoreActions.loadShoes(ourData)_ to send our payload to the _Dispatcher_, which will broadcast it. Then the ShoeStore will “hear” that event and call a method thats loads up some shoes!
+
+`/** @jsx React.DOM */`</br>
```

Comment:
Same as above. Please use triple quote syntax

Reply:
Thanks for guiding 😊 .I will surely make changes in it